resources.tf

```
@@ -0,0 +1,315 @@
+#providers
+provider "aws" {
+	shared_credentials_file  = "C:\\Users\\ThulsiDossK\\.aws\\credentials"
+	region = "${var.region}"
+}
+
+#resources
+resource "aws_vpc" "vpc" {
+  cidr_block = "${var.cidr_vpc}"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+  tags = {
+    "Environment" = "${var.environment_tag}"
+  }
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = "${aws_vpc.vpc.id}"
```

`${}` is used to interpolate a variable inside a string. i.e. construct a string from several variables combined with each other. 

For example `resource_name = "Primary-${aws_vpc.vpc.id}-${var.code}"` 
If we suppose that `aws_vpc.vpc.id = 123` and `var.code = green` then this example will resolve in
`Primary-123-green`

If we don't use the functionality described above, there is no need to use interpolation. Furthermore, latest versions  of terraform don't support interpolation only
```suggestion
  vpc_id = aws_vpc.vpc.id
```